### PR TITLE
Correctly install service as a LaunchDaemon/LaunchAgent depending on user

### PR DESCRIPTION
**Pull request by nre-ableton**
This change cleans up the usage and documentation for `node_exporter_user`. Prior to this change, the documentation for this variable (and the related `node_exporter_group` variable) claim that it is only used on Mac hosts when `node_exporter_install_from_binary` is defined. In fact, these variables are used in the service definitions, which means they are effectively used by all Unix hosts and for both package and binary installations.

In a related manner, our usage of LaunchDaemon/LaunchAgent was incorrect, and this PR now installs the launchd service under the correct directory depending on whether the service is to be run as root or not.

### CODE REVIEW

**Comment by ala-ableton** [comment]
@nre-ableton I'm interested in reviewing this PR, thanks!

**Review by rco-ableton — APPROVED** [review]

File: tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,13 @@
 ---
+# Note that this task is called without superuser privileges, hence it is meant to be a
+# sanity check. We don't want to create the user from this role, because we don't want to
```
Comment (ala-ableton) [inline tasks/main.yml:3]: > We don't want to create the user from this role

Then this should be run in `check_mode`. See https://docs.ansible.com/ansible/latest/collections/ansible/builtin/user_module.html#attributes.

Comment (nre-ableton) [reply tasks/main.yml]: Oh nifty, I didn't know about that! :+1: Fixed in b4fabaa.

**Review by ala-ableton — COMMENTED** [review]
It seems some tests need to be adapted.

**Comment by nre-ableton** [comment]
> It seems some tests need to be adapted.

So the tests fail because the `node_exporter_user` isn't defined for the molecule tests. I've fixed this in c040a64, but molecule tests are still failing locally for me, so I expect they will fail on CI. I think that this is because the `user` task would *modify* the user because we don't specify all user properties that exactly match the `molecule` user, as some of the default values differ from the actual user properties.

But aside from that, eb6df86 (_Assert that node_exporter user already exists_) is quite messy, and I think that we should drop it. As it currently stands, we assert that `node_exporter_user` already exists, but we only require this variable for binary installations. But for binary installations, we actually create the user in `tasks/unix/install-binary.yml`. However, we still configure the service (which is where the user comes in) regardless, so we need to re-think the role of `node_exporter_user`. I'm going to put this PR back into draft mode while I iron this out.

**Comment by nre-ableton** [comment]
Ok, I have reworked this PR in order to improve the user/group handling.

File: README.md [hunk 2]
```diff
@@ -32,7 +32,9 @@ The following role variables are used for `node_exporter`'s service configuratio
 
 - `node_exporter_args`: A list of arguments to pass to the `node_exporter` service, one
   argument per list item.
+- `node_exporter_group`: Group for the to run `node_exporter_user`.
```
Comment (ala-ableton) [inline README.md]: "Group for the to run" is missing a word.

**Review by ala-ableton — COMMENTED** [review]
Just a minor comment

File: README.md (hunk 2 above)
Comment (nre-ableton) [reply README.md]: Fixed in 90bbddb. Please forgive the force-push, I wanted to fix this and deal with the history rewriting at the same time.

File: README.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ The following role variables are used for `node_exporter`'s service configuratio
 
 - `node_exporter_args`: A list of arguments to pass to the `node_exporter` service, one
   argument per list item.
-- `node_exporter_group`: Group for the to run `node_exporter_user`.
+- `node_exporter_group`: Group for `node_exporter_user`.
```
[the remark this answers is not in the document]
Comment (nre-ableton) [reply README.md]: See 830c0e7. Sorry for the force-push (again).

**Review by ala-ableton — APPROVED** [review]

**Comment by nre-ableton** [comment]
I guess we should also bump the role's minor version, similar to what we did with `ableton.jenkins_swarm_client`.

**Comment by ala-ableton** [comment]
> I guess we should also bump the role's minor version, similar to what we did with `ableton.jenkins_swarm_client`.

Yep, as announced in 2968a601e2d88dfd57a05b190a003b4237f7c538's commit message.

**Pull request merged**